fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -348,17 +353,45 @@ public CompletableFuture<Boolean> markIndexReadable(boolean markReadablePlease)
         if (!markReadablePlease) {
```

Comment:
🤷 That's a great question. I've got it as a legacy, can we eliminate this option without breaking backward compatibility?

Reply:
I'm trying to remember the full history here, but it may be a result (ultimately) of either:

1. Wanting to support parallelized index builds (where different parts of the keyspace could be farmed out to different workers, in which case you wouldn't want any of them to try and mark the index as readable until they had all completed), or
2. The option existed because we could but not because there was a use case in mind.

I'm not sure if I recall the full history on that off the top of my head, though.